lib/internal/test_runner/coverage.js

```diff
@@ -485,22 +488,39 @@ function mapRangeToLines(range, lines) {
   return { __proto__: null, lines: mappedLines, ignoredLines };
 }
 
-function shouldSkipFileCoverage(url) {
+function shouldSkipFileCoverage(url, workingDirectory) {
   // The first part of this check filters out the node_modules/ directory
   // from the results. This filter is applied first because most real world
   // applications will be dominated by third party dependencies. The second
   // part of the check filters out core modules, which start with 'node:' in
   // coverage reports, as well as any invalid coverages which have been
   // observed on Windows.
-  return StringPrototypeIncludes(url, '/node_modules/') || !StringPrototypeStartsWith(url, 'file:');
+  if (StringPrototypeIncludes(url, '/node_modules/') || !StringPrototypeStartsWith(url, 'file:')) return true;
+
+  // This check filters out files that match the exclude globs.
+  for (let i = 0; i < excludeFileGlobs.length; ++i) {
+    const absolutePath = fileURLToPath(url);
+    const relativePath = relative(workingDirectory, absolutePath);
+    if (matchesGlob(relativePath, excludeFileGlobs[i]) || matchesGlob(absolutePath, excludeFileGlobs[i])) return true;
```

Comment:
What would you recommend doing instead?

I assume users would want relative checks, but if the path is somewhere completely different, they may also want absolute?

Reply:
So one issue is that you seem to be computing `absolutePath` and `relativePath` for each iteration of the loop, even though I believe they only need to be computed once per function call.

The other thing is, since you already have absolute paths, shouldn't it be enough to glob against those? I guess I don't see anything in the tests added by this PR that would require converting `url` to both an absolute and relative path. This would have a nice side effect of not needing to pass `workingDirectory` around as well. Do you have an example where that wouldn't be the case?

Reply:
I can fix the position of the generation, but here is my reasoning for both relative and absolute testing:
```js
var { minimatch } = require("minimatch")
var path = require("path");

const cwd = "/home/user"
const absTestFile = "/home/user/path/to/file.js"
const relTestFile = path.relative(cwd, absTestFile);

console.log("relTestFile: " + relTestFile + " | Minimatch: " + minimatch(relTestFile, "path/to/*.js")) // true
console.log("absTestFile: " + absTestFile + " | Minimatch: " + minimatch(absTestFile, "path/to/*.js")) // false
```

Users probably want to match the relative test file when excluding coverage files.

Reply:
Ok, fair enough.